index.html

```diff
@@ -652,24 +658,50 @@ <h3>
-          "PerformanceNavigationTiming">unloadEventEnd</a>. If the navigation
-          URL has an <a data-cite="service-workers#dfn-active-worker">active
-          worker registration</a>, immediately before the user agent
-          <a data-cite="service-workers#run-service-worker">runs the worker</a>
-          record the time as {{workerStart}}, or if the worker is available,
-          record the time before the <a data-cite=
-          "service-workers#on-fetch-request-algorithm">event named `fetch` is
-          fired</a> at the active worker. Otherwise, if the navigation URL has
-          no matching <a data-cite=
-          "service-workers#dfn-service-worker-registration">service worker
-          registration</a>, set {{workerStart}} value to zero.
+          "PerformanceNavigationTiming">unloadEventEnd</a>.
+          </li>
+          <li id="worker-start-step">
+            <i>[worker-start-step]</i>
+              <ul>
+                <li>If the [=request/current url=]
+                  has no matching <a data-cite=
+                  "service-workers#dfn-service-worker-registration">service worker
+                  registration</a>, set {{workerStart}} to a {{DOMHighResTimeStamp}}
+                  with a time value of 0, and go to the
+                  <a href="#fetch-start-step">fetch-start-step</a>.</li>
+                <li>
+                  If the [=request/current url=] has an <a data-cite="service-workers#dfn-active-worker">active
+                  worker registration</a>:
+                  <ul>
+                    <li>Let |worker| be the registration's active worker</li>
+                    <li>If |lastWorkerStartOrigin| is not equal to the [=request/current url=]'s origin, set {{workerStart}} to a {{DOMHighResTimeStamp}}
+                      with a time value of 0.</li>
+                    <li>If the user agent has not yet
+                      <a data-cite="service-workers#run-service-worker">run the worker</a>,
+                      immediately before the user agent
+                      <a data-cite="service-workers#run-service-worker">runs the worker</a>:
+                      <ol>
+                        <li>Record the time as {{workerStart}}</dfn>.</li>
+                        <li>Let |lastWorkerStartOrigin| be the [=request/current url=]'s origin.</li>
+                        <li>Go to the <a href="#fetch-start-step">fetch-start-step</a>.</li>
+                      </ol>
+                    </li>
+                    <li>If |lastWorkerStartOrigin| is equal to the [=request/current url=]'s origin, go to the
+                      <a href="#fetch-start-step">fetch-start-step</a>.</li>
+                    <li>If the worker is available, immediately before the <a data-cite=
+                      "service-workers#on-fetch-request-algorithm">event named `fetch` is
+                      fired</a> at the worker, record the time as {{workerStart}}.</li>
```

Comment:
We also need to update `lastWorkerStartOrigin` here

Reply:
Good catch, that would be the case where there is a SW, it's a different origin from last redirect, and the SW is already started (possibly from a a.com->b.com->a.com path).